conans/client/conan_api.py

```diff
@@ -220,8 +222,9 @@ def __init__(self, cache_folder=None, output=None, user_io=None, http_requester=
         # Migration system
         migrator = ClientMigrator(self.cache_folder, Version(client_version), self.out)
         migrator.migrate()
-        # FIXME Remove in Conan 2.0
-        sys.path.append(os.path.join(self.cache_folder, "python"))
+        if not get_env(CONAN_V2_MODE_ENVVAR, False):
+            # FIXME Remove in Conan 2.0
+            sys.path.append(os.path.join(self.cache_folder, "python"))
```

Comment:
It would have been worth to check if the folder is there with contents, and raise, to avoid people using this trying to figure out why it doesn't work. But ok as-is.

Reply:
Yes and no. I would prefer to raise if there is something in that folder, but then you would need to remove the folder in order to try Conan with V2_MODE.... and when you come back to V1 you need to restore that folder.

IMO, testing V2_MODE shouldn't require any modification that will make V1 fail.